packages/yourloops/lib/auth/hook.tsx

```diff
@@ -115,12 +114,12 @@ function AuthContextImpl(api: AuthAPI): AuthContext {
 
     sessionStorage.setItem(STORAGE_KEY_SESSION_TOKEN, auth.sessionToken);
     sessionStorage.setItem(STORAGE_KEY_TRACE_TOKEN, auth.traceToken);
-    sessionStorage.setItem(STORAGE_KEY_USER, JSON.stringify(user));
-    setUser(user);
+    sessionStorage.setItem(STORAGE_KEY_USER, JSON.stringify(auth.user));

```

Comment:
When below we reload the page, the JSON object is parse and use at-is.
Since now it is a class, it's not an object anymore.
This will lead to incoherent code.